python/pyarrow/parquet/core.py

```diff
@@ -1825,7 +1826,15 @@ def read_table(source, *, columns=None, use_threads=True,
         filesystem, path = _resolve_filesystem_and_path(source, filesystem)
         if filesystem is not None:
             source = filesystem.open_input_file(path)
-        # TODO test that source is not a directory or a list
+        if not (
+            (isinstance(source, str) and not os.path.isdir(source))
```

Comment:
What if source is a folder in an S3 bucket? [`os.path.isdir`](https://docs.python.org/3/library/os.path.html#os.path.isdir
will return True for *existing* folder. Will it check through the s3 fs? That would add a network call. Can we just check if the string is a valid folder path?